driver-async/src/test/unit/com/mongodb/async/client/MongoIterableSubscriptionSpecification.groovy

```diff
@@ -325,6 +325,27 @@ class MongoIterableSubscriptionSpecification extends Specification {
         observer.assertUnsubscribed()
     }
 
+    def 'should close the batchCursor when unsubscribe is called prematurely'() {
```

Comment:
I'm also not able to get this test to fail even after reverting the changes in MongoIterableSubscription.  It would be good if we could design a test that fails without the fix applied, and passes with the fix applied

Reply:
Removed.